packages/xod-client/src/editor/components/CppImplementationEditor.jsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import CodeMirror from 'react-codemirror';
+import 'codemirror/mode/clike/clike';
+
+const CppImplementationEditor = ({ isActive, source, isReadOnly, onChange, onClose }) => {
+  const options = {
+    lineNumbers: true,
+    readOnly: isReadOnly,
```

Comment:
Is it really necessary in case that we can edit libraries?
@nkrkv, what will you say? :)

Reply:
Also, in `readOnly` mode it looks the same. And there is no explanation why User can't edit implementation.... Could be recognized as a bug.

Reply:
@evgenykochetkov @brusherru Agree. It’s better to always allow editing.